plugins/experimental/traffic_dump/traffic_dump.cc

```diff
@@ -420,36 +568,39 @@ global_ssn_handler(TSCont contp, TSEvent event, void *edata)
       tag.remove_prefix(PLUGIN_PREFIX.size());
       if (tag == "sample") {
         sample_pool_size = static_cast<int64_t>(strtol(static_cast<const char *>(msg->data), nullptr, 0));
-        TSDebug(PLUGIN_NAME, "TS_EVENT_LIFECYCLE_MSG: Received Msg to change sample size to %" PRId64 "bytes",
+        TSDebug(PLUGIN_NAME, "TS_EVENT_LIFECYCLE_MSG: Received Msg to change sample size to %" PRId64 " bytes",
                 sample_pool_size.load());
       } else if (tag == "reset") {
         disk_usage = 0;
         TSDebug(PLUGIN_NAME, "TS_EVENT_LIFECYCLE_MSG: Received Msg to reset disk usage counter");
       } else if (tag == "limit") {
         max_disk_usage = static_cast<int64_t>(strtol(static_cast<const char *>(msg->data), nullptr, 0));
-        TSDebug(PLUGIN_NAME, "TS_EVENT_LIFECYCLE_MSG: Received Msg to change max disk usage to %" PRId64 "bytes",
+        TSDebug(PLUGIN_NAME, "TS_EVENT_LIFECYCLE_MSG: Received Msg to change max disk usage to %" PRId64 " bytes",
                 max_disk_usage.load());
       }
     }
     return TS_SUCCESS;
   }
   case TS_EVENT_HTTP_SSN_START: {
     // Grab session id to do sampling
-    int64_t id = TSHttpSsnIdGet(ssnp);
+    int64_t id                = TSHttpSsnIdGet(ssnp);
+    const sockaddr *client_ip = TSHttpSsnClientAddrGet(ssnp);
     if (id % sample_pool_size != 0) {
-      TSDebug(PLUGIN_NAME, "global_ssn_handler(): Ignore session %" PRId64 "...", id);
+      TSDebug(PLUGIN_NAME, "%s: Ignore session %" PRId64 " per the random sampling mechanism", __func__, id);
       break;
     } else if (disk_usage >= max_disk_usage) {
-      TSDebug(PLUGIN_NAME, "global_ssn_handler(): Ignore session %" PRId64 "due to disk usage %" PRId64 "bytes", id,
-              disk_usage.load());
+      TSDebug(PLUGIN_NAME, "%s: Ignore session %" PRId64 " due to disk usage %" PRId64 " bytes", __func__, id, disk_usage.load());
+      break;
+    } else if (is_filtered_out(client_ip)) {
+      TSDebug(PLUGIN_NAME, "%s: Ignore session %" PRId64 " per the client's IP filter", __func__, id);
       break;
     }
     // Beginning of a new session
     /// Get epoch time
     auto start = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::system_clock::now().time_since_epoch());
 
     // Create new per session data
-    SsnData *ssnData = new SsnData;
+    SsnData *ssnData = new SsnData();
```

Comment:
Prefer new style initializer if possible, e.g. `SsnData{}`. In this case, though, I wouldn't make the change at all.